src/huggingface_hub/utils/_git_credential.py

```diff
@@ -37,9 +37,15 @@ def list_credential_helpers(folder: Optional[str] = None) -> List[str]:
         # NOTE: If user has set an helper for a custom URL, it will not we caught here.
         #       Example: `credential.https://huggingface.co.helper=store`
         #       See: https://github.com/huggingface/huggingface_hub/pull/1138#discussion_r1013324508
+        for line in output.split("\n"):
+            if "credential.helper" in line:
+                if line.split("=")[-1]:
+                    value = line.split("=")[-1].split()[0]
+                else:
+                    value = line.split("=")[-1].strip()
```

Comment:
Here I think the parsing will break if line = ` credential.helper=  ` since `line.split("=")[-1]` will evaluate to True but `line.split("=")[-1].split()[0]` will fail.

Would you like to try to fix this once and for all using a regex? I started something [here](https://regex101.com/r/dSmWyB/1) that in theory should work for all the cases we want to handle (still has to be tested for real).

![image](https://github.com/huggingface/huggingface_hub/assets/11801849/4489e80d-8740-41ef-8869-cb6a75f5f79f)

Reply:
Sure, I'll re-evaluate the parsing.

Regarding the usage of Regex,
It sounds more robust approach, many thanks for putting down the sample regex and sharing it with me. I'll test it for other potential use cases and incorporate the same in my ongoing PR